source/m2mbase.cpp

```diff
@@ -354,17 +354,26 @@ M2MBase::Operation M2MBase::operation() const
 
 const char* M2MBase::name() const
 {
-    return _sn_resource->name;
+    assert(_sn_resource->identifier_int_type == false);
+    return _sn_resource->identifier.name;
 }
 
 int32_t M2MBase::name_id() const
 {
-    return _sn_resource->name_id;
+    int32_t name_id = -1;
+    if(is_integer(_sn_resource->identifier.name) && strlen(_sn_resource->identifier.name) <= MAX_ALLOWED_STRING_LENGTH) {
```

Comment:
This should have a "assert(_sn_resource->identifier_int_type == false)" or just handle that case here also, as now the code may use the instance_id as a string pointer with random results.